### PR TITLE
OPEN-86 Implemented an instance mapper in ran

### PR DESCRIPTION
**Pull request by anton-larsen-persequor**
This is intended as a framework from mapping from one representation of an object to another.

For example it could be used to map from an EPCIS event to an industry specific message type.

### CODE REVIEW

File: ran-core/src/main/java/io/ran/MapperGenerator.java [hunk 1]
```diff
@@ -16,7 +16,7 @@ public Wrapped generate(AutoMapperClassLoader classLoader, Clazz clazz) {
 
 			MappingClassWriter visitor = new MappingClassWriter(clazz.clazz);
 			byte[] bytes = visitor.toByteArray();
-//			Files.write(path, bytes);
+			Files.write(path, bytes);
```
Comment (MikePSQR) [inline ran-core/src/main/java/io/ran/MapperGenerator.java:19]: debug

**Comment by MikePSQR** [comment]
https://persequor.atlassian.net/browse/OPEN-82

File: ran-core/src/main/java/io/ran/MapperGenerator.java (hunk 1 above)
Comment (emanuel-persequor) [inline ran-core/src/main/java/io/ran/MapperGenerator.java:19]:
```suggestion
			//Files.write(path, bytes);
```

File: ran-class-mapper/src/main/java/io/ran/instancemapper/InstanceMappingRegistry.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+import io.ran.token.Token;
+
+import javax.inject.Inject;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+public class InstanceMappingRegistry {
+    Map<Class<?>, Map<Class<?>, Map<Class<?>, Map<Token, InstanceFieldMapper>>>> classSpecificMappings = new ConcurrentHashMap<>();
```
Comment (MikePSQR) [inline ran-class-mapper/src/main/java/io/ran/instancemapper/InstanceMappingRegistry.java:13]: should we use CacheTriple instead of 3 maps?

Comment (MikePSQR) [reply ran-class-mapper/src/main/java/io/ran/instancemapper/InstanceMappingRegistry.java]: yeah I did

**Review by niels-psqr — APPROVED** [review]
This seems to be relatively narrow in scope, but its also limited to its own project, so I don't see any harm in including it. :+1:

**Pull request merged**